service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java

```diff
@@ -66,10 +70,23 @@ public void createRecordType(UUID instanceId, Map<String, DataTypeMapping> table
 			}
 			throw e;
 		}
+
 	}
 
 	private String getQualifiedTableName(String recordType, UUID instanceId) {
-		return quote(instanceId.toString()) + "." + quote(recordType);
+		return quote(instanceId.toString()) + "." + quote(validateName(recordType, InvalidNameException.NameType.RECORD_TYPE));
+	}
+
+	private String validateName(String name, InvalidNameException.NameType nameType){
+		if(containsDisallowedSqlCharacter(name)){
+			throw new InvalidNameException(nameType);
+		} else {
+			return name;
+		}
```

Comment:
nit: the `else` is extraneous here:
```suggestion
		if(containsDisallowedSqlCharacter(name)){
			throw new InvalidNameException(nameType);
		}
		return name;
```
no big deal if you have a reason to keep it as-is.